parsers/AR.py

```diff
@@ -0,0 +1,653 @@
+    nuclear_generation = sum([mapped_data[i] for i in find_nuclear])
+    find_oil = [i+2 for i, x in enumerate(mapped_data) if x == 'oil']
+    oil_generation = sum([mapped_data[i] for i in find_oil])
+    find_coal = [i+2 for i, x in enumerate(mapped_data) if x == 'coal']
+    coal_generation = sum([mapped_data[i] for i in find_coal])
+    find_biomass = [i+2 for i, x in enumerate(mapped_data) if x == 'biomass']
+    biomass_generation = sum([mapped_data[i] for i in find_biomass])
+    find_gas = [i+2 for i, x in enumerate(mapped_data) if x == 'gas']
+    gas_generation = sum([mapped_data[i] for i in find_gas])
+  
+    unknown_generation = (thermal_generation - nuclear_generation - gas_generation \
+                          - oil_generation - coal_generation - biomass_generation)
+
+    return {'gas': gas_generation, 
+            'nuclear': nuclear_generation,
+            'coal': coal_generation,
+            'unknown': unknown_generation,
+            'oil': oil_generation,
+            'biomass': biomass_generation
+           }
+
+
+
+def get_hydro():
+    """Requests hydro generation data then parses, returns a dictionary."""
+
+    s = requests.Session()
+    r = s.get(hurl)
+    pat = re.search("ControlID=[^&]*", r.text).group()
+    spat = re.search("ReportSession=[^&]*", r.text).group()
+    cid = pat.rpartition('=')[2]
+    rs = spat.rpartition('=')[2]
+    full_table = []
+    
+
+    for pagenumber in range(1,2,1):
+        t = s.get(thurl, params = {'ControlID': cid, 'ReportSession': rs, 'PageNumber': '{}'.format(pagenumber)})
+        text_only = webparser(t)
+        if 'En Reserva' in text_only:
+            break
+        full_table.append(text_only)
+
+    data = list(itertools.chain.from_iterable(full_table))
+    formatted_data = dataformat(data)
+    find_hydro = [i+1 for i,x in enumerate(formatted_data) if x == 'Totales ']
+    total_hydro_generation = sum([formatted_data[i] for i in find_hydro])
+
+    return {'hydro': total_hydro_generation}
+
+
+
+def fetch_production(country_code='AR'):
+    """
+    Requests the last known production mix (in MW) of a given country
+    Arguments:
+    country_code (optional) -- used in case a parser is able to fetch multiple countries
+    Return:
+    A dictionary in the form:
+    {
+      'countryCode': 'FR',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
+      },
+      'storage': {
+          'hydro': -10.0,
+      },
+      'source': 'mysource.com'
+    }
+    """
+    gdt = get_datetime()
+    thermal = get_thermal()
+    hydro = get_hydro()
+    production_mix = {
+      'countryCode': country_code,
+      'datetime': gdt['datetime'],
```

Comment:
The datetime value needs to be a python datetime.
You can change L487 `.format('YYYY-MM-DD HH:mm:ss')` to `.datetime`